flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/ResultPartitionBuilder.java

```diff
@@ -40,6 +40,8 @@
 
 	private BoundedBlockingSubpartitionType blockingSubpartitionType = BoundedBlockingSubpartitionType.AUTO;
 
+	private int partitionIndex = 0;
```

Comment:
I think it's better to initialize it with an invalid value to force clients to set it.

Reply:
This builder is only for testing purpose, and I do not think this property is more important or very necessary than other existing fields like `partitionId`, `numberOfSubpartitions` etc. Even I guess this new field is rare used for most of the testing cases, so I prefer to not making it as mandatory.